File: go/binary_heap/heap_sort_test.go
package binary_heap

import (
	"github.com/stretchr/testify/assert"
	"reflect"
	"sort"
	"testing"
)

func TestHeapSort_RandomCases(t *testing.T) {
	for range 1_000 {
		td := newRandomTestData()

		expected := append([]int{}, td.array...)
		sort.Ints(expected)

		actual := append([]int{}, td.array...)
		HeapSort(actual)

		if !reflect.DeepEqual(actual, expected) {
			assert.FailNowf(t, "",
				"HeapSort(%v):\nwant %v\n got %v", td.array, expected, actual,
			)
		}
	}
}

File: python/binary_heap/heap_sort_test.py
from bh_random_test_data import RandomTestData
from heap_sort import heap_sort


def test_heap_sort_random_cases():
	for _ in range(1_000):
		td = RandomTestData()

		want = sorted(td.array)

		got = td.array.copy()
		heap_sort(got)

		assert got == want, \
			f'HeapSort({td.array}):\nwant {want}\n got {got}'
